src/lib/incentive-relationship-calculation.ts

```diff
@@ -93,25 +122,50 @@ export function buildRelationshipGraph(data: IncentiveRelationships) {
   return edges;
 }
 
-// Checks the prerequisites for a single incentive.
-export function meetsPrerequisites(
+// Helper to traverse nested prerequisites and check whether they are met for
+// the given incentive.
+function meetsSinglePrerequisite(
```

Comment:
I don't love the name simply because it doesn't feel like there is anything more "single" about this than meetsPrerequisites... both of them are called on a single incentiveID.

Maybe meetsPrerequisiteGraph or satisfiesPrerequisiteGraph? #namingishard

Reply:
Naming is indeed the worst. And actually even though this is talking about a single prereq object, it is really checking the full tree of prerequisites. How about "meetsNestedPrerequisites"?